### PR TITLE
Test cl_ext_cxx_for_opencl (#1075)

### PR DESCRIPTION
**Pull request by AnastasiaStulova**
Add tests for API extension for compilation
of kernels in C++ for OpenCL language.
* Test that -cl-std=CLC++ is accepted and a
  basic kernel with C++ features is compiled.
* Test that API extension reports the same
  language version as __OPENCL_CPP_VERSION__.

This commit also adds a separate folder for extension
tests.

Patch by Victoria Holodovsky!

### CODE REVIEW

File: test_conformance/extensions/cl_ext_cxx_for_opencl/cxx_for_opencl_ext.cpp [hunk 1]
```diff
@@ -0,0 +1,103 @@
+//
+// Copyright (c) 2020 The Khronos Group Inc.
```
Comment (AnastasiaStulova) [inline test_conformance/extensions/cl_ext_cxx_for_opencl/cxx_for_opencl_ext.cpp:2]: The copyright year needs to change!

File: test_conformance/extensions/cl_ext_cxx_for_opencl/cxx_for_opencl_ext.cpp [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+
+    error = create_single_kernel_helper_with_build_options(
+        context, &program, &kernel1, 1, &kernel_sstr, "k1",
+        "-cl-std=CLC++", false);
+    test_error(error, "Failed to create k1 kernel");
+
+    kernel2 = clCreateKernel( program, "k2", &error );
+    test_error(error, "Failed to create k2 kernel");
+
+    in_buffer = clCreateBuffer(context, CL_MEM_READ_WRITE | CL_MEM_COPY_HOST_PTR, sizeof(value), &value,
```
Comment (AnastasiaStulova) [inline test_conformance/extensions/cl_ext_cxx_for_opencl/cxx_for_opencl_ext.cpp:53]: The formating is different to the line 45 btw.

@kpet do we run clang-format or something?

File: test_conformance/extensions/cl_ext_cxx_for_opencl/cxx_for_opencl_ext.cpp [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+#include "procs.h"
+
+
+int test_cxx_for_opencl(cl_device_id device, cl_context context,
+                        cl_command_queue queue, int num_elements,
+                        bool useCoreSubgroups)
```
Comment (kpet) [inline test_conformance/extensions/cl_ext_cxx_for_opencl/cxx_for_opencl_ext.cpp:21]: Some of the parameters are not necessary (at least `num_elements` and `useCoreSubgroups`).

**Review by kpet — CHANGES_REQUESTED** [review]
It would be nice to credit the original author of those tests. I suggest adding a Signed-off-by and maybe resetting the commit author as well so they're credited in the git history too.

**Comment by AnastasiaStulova** [comment]
> It would be nice to credit the original author of those tests. I suggest adding a Signed-off-by and maybe resetting the commit author as well so they're credited in the git history too.

Ok I have updated the first (i.e. base) commit.

**Comment by AnastasiaStulova** [comment]
> > It would be nice to credit the original author of those tests. I suggest adding a Signed-off-by and maybe resetting the commit author as well so they're credited in the git history too.
> 
> Ok I have updated the first (i.e. base) commit.

This didn't work, so I have repushed it now. But there is an issue with CLA now because I think @VictoriaHolodovsky needs to sign it too?

**Comment by AnastasiaStulova** [comment]
> > > It would be nice to credit the original author of those tests. I suggest adding a Signed-off-by and maybe resetting the commit author as well so they're credited in the git history too.
> > 
> > 
> > Ok I have updated the first (i.e. base) commit.
> 
> This didn't work, so I have repushed it now. But there is an issue with CLA now because I think @VictoriaHolodovsky needs to sign it too?

the CLA has been signed!

**Review by kpet — APPROVED** [review]

**Review by gwawiork — APPROVED** [review]
LGTM

**Pull request merged**